build/source/dshare/outpt_stat.f90

```diff
@@ -132,71 +137,75 @@ subroutine calc_stats(meta,stat,tdata,iStep,err,message)
-  select case(iStat)                               ! act depending on the statistic
-   case (iLookStat%totl)                           ! summation over period
-    tstat(iStat) = tstat(iStat) + tdata            ! into summation
-   case (iLookStat%inst)                           ! instantaneous
-    tstat(iStat) = tdata
-   case (iLookStat%mean)                           ! mean over period
-    tstat(iStat) = tstat(iStat) + tdata            ! adds timestep to sum
-   case (iLookStat%vari)                           ! variance over period
-    tstat(iStat) = tstat(iStat) + tdata**2         ! sum into E[X^2] term
-    tstat(maxVarStat+1) = tstat(maxVarStat+1) + tdata  ! sum into E[X]^2 term
-   case (iLookStat%mini)                           ! minimum over period
-    if (tdata.le.tstat(iStat)) tstat(iStat) = tdata! overwrites minimum iff
-   case (iLookStat%maxi)                           ! maximum over period
-    if (tdata.ge.tstat(iStat)) tstat(iStat) = tdata! overwrites maximum iff
-   case (iLookStat%mode)                           ! (does not work)
-    tstat(iStat) = -9999.
+ do iFreq=1,maxVarFreq                                ! loop through output statistics
+  if(meta%statIndex(iFreq)==integerMissing) cycle     ! don't bother if output frequency is not desired for a given variab;e
+  if(meta%varType/=iLookVarType%outstat) cycle        ! only calculate stats for scalars
+  select case(meta%statIndex(iFreq))                  ! act depending on the statistic
+   ! -------------------------------------------------------------------------------------
+   case (iLookStat%inst)                              ! * instantaneous value 
+    tstat(iFreq) = tdata                              !     - data at a given time
+   case (iLookStat%totl)                              ! * summation over period                    
+    tstat(iFreq) = tstat(iFreq) + tdata*data_step     !     - increment data 
+   case (iLookStat%mean)                              ! * mean over period                       
+    tstat(iFreq) = tstat(iFreq) + tdata               !     -  increment data
+   case (iLookStat%vari)                              ! * variance over period                   
+    tstat(iFreq) = tstat(iFreq) + tdata**2                     ! - E[X^2] term in var calc    
+    tstat(maxVarFreq+iFreq) = tstat(maxVarFreq+iFreq) + tdata  ! - E[X]^2 term                 
+   case (iLookStat%mini)                              ! * minimum over period                    
+    if (tdata<tstat(iFreq)) tstat(iFreq) = tdata      !     - check value 
+   case (iLookStat%maxi)                              ! * maximum over period                    
+    if (tdata>tstat(iFreq)) tstat(iFreq) = tdata      !     - check value 
+   case (iLookStat%mode)                              ! * mode over period (does not work)       
+    tstat(iFreq) = realMissing
+   ! -------------------------------------------------------------------------------------
```

Comment:
Can you add a default to trap any errors (it should never get there, but coding errors happen and this would be easier to catch them)?

Reply:
Done